src/components/modals/modalJSONViewer/index.tsx

```diff
@@ -6,36 +6,42 @@ import {
   Modal,
 } from '../modal';
 
-import type { IMappedBlockExtrinsic } from '@custom-types/block';
+import type {
+  IMappedBlockEvent,
+  IMappedBlockExtrinsic,
+} from '@custom-types/block';
 
 interface IModalJSONViewer extends Pick<IModal, 'onClose'> {
-  extrinsic?: IMappedBlockExtrinsic;
+  jsonData?: IMappedBlockExtrinsic | IMappedBlockEvent | unknown;
+  title?: string;
   onClose: () => void;
 }
 
 export const ModalJSONViewer = (props: IModalJSONViewer) => {
   const {
-    extrinsic,
+    jsonData,
     onClose,
+    title,
   } = props;
 
-  if (!extrinsic) {
+  if (!jsonData) {
```

Comment:
it's better to open the modal in terms of UX ... we can use jsonData = {}; or show some message like ... no JSON or something.